great_expectations/rule_based_profiler/domain_builder/cardinality_checker.py

```diff
@@ -0,0 +1,211 @@
+import abc
+import enum
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+from great_expectations.exceptions import ProfilerConfigurationError
+
+
+@dataclass
```

Comment:
@anthonyburdi Some docstrings about what these classes are for would be nice...  Thanks!